src/main/java/core/modules/web/models/Web.java

```diff
@@ -38,76 +38,161 @@ public Web openAnyUrl(String url) {
+    public <T extends Enum> WebElement findElementByCss(T css) {
+        try {
+            return driver.findElement(By.cssSelector(String.valueOf(css)));
+        } catch (Exception e) {
+            Assert.fail("Element with xpath + " + css + "not found");
+            return null;
+        }
+    }
+
+    public WebElement findElementByCss(String css) {
         try {
             return driver.findElement(By.cssSelector(css));
         } catch (Exception e) {
-            Assert.fail("Element with css + " + css + "not found" );
+            Assert.fail("Element with css + " + css + "not found");
             return null;
         }
     }
 
-    public CustomWebElement useElementByLinkText (String text) {
+    public CustomWebElement useElementByLinkText(String text) {
         WebElement element = driver.findElement(By.linkText(text));
-    return new CustomWebElement(element, this, driver);
+        return new CustomWebElement(element, this, driver);
     }
 
-    public Web switchToNewTab () {
-        for(String winHandle : driver.getWindowHandles()){
+    public Web switchToNewTab() {
+        for (String winHandle : driver.getWindowHandles()) {
             driver.switchTo().window(winHandle);
         }
         return this;
     }
 
+    public <T extends Enum> WebAssertion useCssElementForAssertion(T css) {
+        WebElement element = driver.findElement(By.cssSelector(String.valueOf(css)));
+        return new WebAssertion(element);
+    }
+
     public WebAssertion useCssElementForAssertion(String css) {
-        WebElement element = findElementByCss(css);
+        WebElement element = driver.findElement(By.cssSelector(String.valueOf(css)));
         return new WebAssertion(element);
     }
 
+    public <T extends Enum> Web useCssElementForAssertionAnd(T css) {
+        WebElement element = driver.findElement(By.cssSelector(String.valueOf(css)));
+        return this;
+    }
+
     public Web useCssElementForAssertionAnd(String css) {
         WebElement element = findElementByCss(css);
         return this;
     }
 
+    public <T extends Enum> WebAssertion useXpathElementForAssertion(T xpath) {
+        WebElement element = driver.findElement(By.xpath(String.valueOf(xpath)));
+        return new WebAssertion(element);
+    }
+
     public WebAssertion useXpathElementForAssertion(String xpath) {
         WebElement element = findElementByXpath(xpath);
         return new WebAssertion(element);
     }
 
+    public <T extends Enum> Web useXpathElementForAssertionAnd(T xpath) {
+        WebElement element = driver.findElement(By.xpath(String.valueOf(xpath)));
+        return this;
+    }
+
     public Web useXpathElementForAssertionAnd(String xpath) {
         WebElement element = findElementByXpath(xpath);
         return this;
     }
 
+    public <T extends Enum> WebAssertion useXpathElementsForAssertion(T xpath) {
+        List<WebElement> elements = driver.findElements(By.xpath(String.valueOf(xpath)));
```

Comment:
Is this case that you were telling about? when we have a list of elements and want to use one of them

Reply:
yes, that is correct. Sometimes we will need the collection of web elements to interact with.

Reply:
So I created these methods for almost all existing cases.